Add provider config management operations.

micahstairs opened the pull request:
This adds all of the OIDC and SAML provider config operations, as part of adding multi-tenancy support.

hiranya911 commented:
@kevinthecheung can you please review the API docs in following classes:

* AbstractFirebaseAuth
* ProviderConfig
* SamlProviderConfig
* OidcProviderConfig
* ListProviderConfigsPage

kevinthecheung commented on src/main/java/com/google/firebase/auth/AbstractFirebaseAuth.java:
```diff
@@ -945,6 +939,543 @@ protected String execute() throws FirebaseAuthException {
+  }
+
+  /**
+   * Gets the SAML provider Auth config corresponding to the specified provider ID.
```
auth provider

micahstairs replied:
Done.

kevinthecheung commented on src/main/java/com/google/firebase/auth/OidcProviderConfig.java:
```diff
@@ -0,0 +1,180 @@
+     * provider.
+     *
+     * <p>The returned object should be passed to
+     * {@link AbstractFirebaseAuth#updateOidcProviderConfig(CreateRequest)} to update the provider
```
to save the updated config

micahstairs replied:
Done.

micahstairs commented on src/main/java/com/google/firebase/auth/AbstractFirebaseAuth.java:
```diff
@@ -945,6 +939,543 @@ protected String execute() throws FirebaseAuthException {
   }
 
+  /**
+   * Creates a new OIDC Auth provider config with the attributes contained in the specified
```
Done.

micahstairs commented on src/main/java/com/google/firebase/auth/AbstractFirebaseAuth.java:
```diff
@@ -945,6 +939,543 @@ protected String execute() throws FirebaseAuthException {
+  }
+
+  /**
+   * Gets the provider OIDC Auth config corresponding to the specified provider ID.
```
Done.

micahstairs commented on src/main/java/com/google/firebase/auth/AbstractFirebaseAuth.java:
```diff
@@ -945,6 +939,543 @@ protected String execute() throws FirebaseAuthException {
+
+  /**
+   * Similar to {@link #getOidcProviderConfig(String)} but performs the operation asynchronously.
+   * Page size will be limited to 100 provider configs.
```
Done.

micahstairs commented:
Alright, thanks for the review! I've addressed your feedback, and I've also rebased this branch so that we don't have any more merge conflicts with `tenant-mgt`.
